homeassistant/components/automation/__init__.py

```diff
@@ -510,19 +504,25 @@ async def async_trigger(self, run_variables, context=None, skip_condition=False)
                 variables = run_variables
             automation_trace.set_variables(variables)
 
+            # Wipe the trace, and set the new trace as the automation's condition trace
```

Comment:
Why do we wipe it? Can we include that in the comment

Reply:
Better wording would be:
`# Reset tracing, and set the new trace as the automation's condition trace`

Is it clearer?

Reply:
Why do we reset it? What was the old one about and where is it now?

Reply:
We clear all the context variables, including setting `trace_cv = {}`, meaning we start a new trace. We want one for the conditions and one for the actions.

Reply:
I updated the comment to try to make it less confusing.